rtgui/editbuffer.cc

```diff
@@ -108,9 +108,11 @@ int ObjectMOBuffer::getObjectID(const rtengine::Coord& location)
     }
 
     if (objectMode == OM_255) {
-        id = (unsigned char)(*( objectMap->get_data() + location.y * objectMap->get_stride() + location.x ));
+        // In OM_255 mode, size of pixel is 1 Byte (i.e. size of unsigned char)
+        memcpy(&id, ( objectMap->get_data() + location.y * objectMap->get_stride() + sizeof(unsigned char) * location.x ), sizeof(unsigned char));
     } else {
-        id = (unsigned short)(*( objectMap->get_data() + location.y * objectMap->get_stride() + location.x ));
+        // In OM_65535 mode, size of pixel is 2 Bytes (i.e. size on unsigned short)
+        memcpy(&id, ( objectMap->get_data() + location.y * objectMap->get_stride() + sizeof(unsigned short) * location.x ), sizeof(unsigned short));
```

Comment:
If the size of the type is important, isn't it better to use [fixed width types](https://en.cppreference.com/w/cpp/types/integer) like `std::uint8_t` and `std::uint16_t`?